Shared/Playback/Models/Track.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  Track.swift
+//  Relisten
+//
+//  Created by Alec Gorge on 7/3/17.
+//  Copyright © 2017 Alec Gorge. All rights reserved.
+//
+
+import Foundation
+
+public class Track : Codable, Hashable {
+    public enum PlaybackState {
+        case notActive
+        case paused
+        case playing
+    }
+
+    public enum DownloadState {
+        case none
+        case queued
+        case downloading
+        case downloaded
+    }
+    
+    public let showInfo : CompleteShowInformation
+    
+    public var id : Int { get { return sourceTrack.id } }
```

Comment:
I like all the changes aside from flattening these property names from the tracks.

I got into a real maintence nightmare in the old Relisten by diverging so much from the API while trying to maintain all the platforms. It isn’t a Swift/Apple style but it helps a lot with API refactors and reduces errors when manipulating the models consistently

Reply:
Would you be ok with it if I kept the same names as the `SourceTrack` so that it's a complete passthrough? I just found it confusing that the top level Track didn't contain all the properties that it really exposed.

Reply:
I pushed the change I was talking about above because Xcode made refactoring this really simple. It's not what your original comment asked for though, so let me know if you'd still like me to just expose `SourceTrack` for the backing properties.

Reply:
I'm okay with it just matching the names